pkg/ext-proc/backend/pod_metrics_refresher.go

```diff
@@ -0,0 +1,79 @@
+package backend
+
+import (
+	"context"
+	"time"
+
+	klog "k8s.io/klog/v2"
+)
+
+const (
+	// TODO: make it configurable
+	fetchMetricsTimeout = 1 * time.Second
+)
+
+type PodMetricsRefresher struct {
+	done     chan struct{}
+	interval time.Duration
+
+	// refresher holds provider & pod so it can update the metrics concurrent-safely.
+	pod      Pod
+	provider *Provider
```

Comment:
It's quite confusing to have this "circular dependency" between the provider and the refresher. My idea is to abstract away the metric refreshing logic from the provider, and move it into the Pod object. However this would require some decent refactoring. I am OK with doing this refactor as a follow up. Feel free to reach out to me on Slack if you would like to do the refactoring.

Reply:
I will open an issue for the refactor once this PR is merged.